packages/vite/src/node/optimizer/index.ts

```diff
@@ -237,12 +238,29 @@ export async function optimizeDeps(
   const idToExports: Record<string, ExportsData> = {}
   const flatIdToExports: Record<string, ExportsData> = {}
 
+  const { plugins = [], ...esbuildOptions } =
+    config.optimizeDeps?.esbuildOptions ?? {}
+
   await init
   for (const id in deps) {
     const flatId = flattenId(id)
     flatIdDeps[flatId] = deps[id]
     const entryContent = fs.readFileSync(deps[id], 'utf-8')
-    const exportsData = parse(entryContent) as ExportsData
+    let exportsData: ExportsData
+    try {
+      exportsData = parse(entryContent) as ExportsData
+    } catch {
+      debug(
+        `Unable to parse dependency: ${id}. Trying again with a JSX transform.`
+      )
+      const transformed = await transform(entryContent, {
```

Comment:
Please use `transformWithEsbuild` instead
https://github.com/vitejs/vite/blob/ed16488591b7dfac7e54d96f67c6f3674368b0cc/packages/vite/src/node/plugins/esbuild.ts#L82

Reply:
Done, apologies for the delay, been a busy week :)